### PR TITLE
Feature/artifacts deployment

### PR DESCRIPTION
**Pull request by stmh**
Hi @d34dman 

this MR contains the changes for doing artifact-based deployments into another git repository. Please have a look into the code and documentation.

### CODE REVIEW

File: docs/deploying-artifacts.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+* `runDeployScript` run the deploy script of the host-config.
+* `pushToTargetRepository` commit and push all changes, using the changelog as a commit-message
+
+If you run phabalicious as part of a CI-setup, it might make sense to set `useLocalRepository` to true, as this will instruct phab to use the current folder as a base for the artifact and won't create a new application in another temporary folder.
+
+### Example config
+
+```yaml
+excludeFiles:
+  ftpSync:
```
Comment (d34dman) [inline docs/deploying-artifacts.md]: Another type, should be gitSync: i believe.

**Comment by d34dman** [comment]
```
hosts:
  sample-sync-host:
    appCreateCopyFilesToTargetDirectoryPrepare:
      - cd %context.data.installDir%; cp .env.example .env
    ignoreSubmodules: true
    target:
      branch: int
      repository: ssh://git@source.factorial.io:2222/skakanat/mi009-phab-artifact.git
    actions:
      - action: copy
        arguments:
          - sites/all/modules
          - sites/all/libraries
          - sites/all/themes
      - action: exclude
        arguments:
          - sites/all/modules/custom/grv-drupal
      - action: delete
        arguments:
          - .tools/es
          - .idea
    needs:
      - git
      - artifacts--git
      - script
```
Update:
Moved `actions` key one level up. (instead of being a key under target).

**Comment by stmh** [comment]
i'd vote to move actions into the target section as they are only valid for that target.

**Comment by stmh** [comment]
and i would rename target to artifact to make sure what this block is about

**Comment by d34dman** [comment]
@stmh  good job :) I have tested this and works. Did a small modification to conveniently copy the files to desired folder. Please merge whenever you like :) Thanks a LOT!

**Pull request merged**